src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java

```diff
@@ -8,45 +8,145 @@
+import com.wikia.webdriver.elements.mercury.pages.ArticlePage;
+import com.wikia.webdriver.elements.mercury.pages.SearchResultsPage;
 
 import org.testng.annotations.Test;
 
 @Execute(onWikia = MercuryWikis.MERCURY_AUTOMATION_TESTING)
 @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+@Test(groups = "MercurySearch")
 public class SearchTests extends NewTestTemplate {
 
-  private TopBar topBar;
-  private Search search;
+  private static final String SEARCH_PHRASE = "Infobox";
+  private static final String SEARCH_PHRASE_NO_RESULTS = "AComplexQueryWithNoResults";
 
-  private void init() {
-    this.topBar = new TopBar(driver);
-    this.search = new Search();
+  @Test(groups = "mercury_search_navigateToPageUsingSearchSuggestions")
+  public void mercury_search_navigateToPageUsingSearchSuggestions() {
+    String suggestionLink =
+        new ArticlePage()
+            .open(MercurySubpages.MAIN_PAGE)
+            .getTopBar()
+            .openSearch()
+            .typeInSearch(this.SEARCH_PHRASE)
+            .selectSearchSuggestion(0);
 
-    new Navigate().toPage(MercurySubpages.MAIN_PAGE);
+    Assertion.assertTrue(driver.getCurrentUrl().equals(suggestionLink));
   }
 
-  @Test(groups = "mercury_search_navigateToPageUsingSearch")
-  public void mercury_search_navigateToPageUsingSearch() {
-    String searchingPhrase = "Infobox";
-    init();
+  @Test(groups = "mercury_search_navigateToPageUsingSearchResults")
+  public void mercury_search_navigateToPageUsingSearchResults() {
+    String resultLink =
+        new SearchResultsPage()
+            .openForQuery(this.SEARCH_PHRASE)
+            .selectSearchResult(0);
 
-    topBar.openSearch().typeInSearch(searchingPhrase).selectSearchSuggestion(0);
-
-    Assertion.assertTrue(driver.getCurrentUrl().contains(searchingPhrase));
+    Assertion.assertTrue(driver.getCurrentUrl().equals(resultLink));
   }
 
   @Test(groups = "mercury_search_cancelSearchPhrase")
   public void mercury_search_clearSearchPhrase() {
-    String searchingPhrase = "Infobox";
-    init();
+    String currentPhrase = new ArticlePage()
+        .open(MercurySubpages.MAIN_PAGE)
+        .getTopBar()
+        .openSearch()
+        .typeInSearch(this.SEARCH_PHRASE)
+        .clickClearSearchButton()
+        .getSearchPhrase();
+
+    Assertion.assertTrue(currentPhrase.isEmpty());
+  }
+
+  @Test(groups = "mercury_search_verifySearchLayout")
+  public void mercury_search_verifySearchLayout() {
+    Search search =
+        new ArticlePage()
+            .open(MercurySubpages.MAIN_PAGE)
+            .getTopBar()
+            .openSearch();
+
+    Assertion.assertTrue(search.isSearchInputFieldVisible());
+    Assertion.assertTrue(search.isClearSearchButtonVisible());
+    Assertion.assertTrue(search.isInputFieldSearchIconVisible());
+  }
+
+  @Test(groups = "mercury_search_userIsRedirectedToSearchResultsPage")
+  public void mercury_search_userIsRedirectedToSearchResultsPage() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS00100)